cpp/src/sort/sort.cuh

```diff
@@ -0,0 +1,70 @@
+// -*-c++-*-
+
+#pragma once
+
+#include <cudf/types.h>
+
+namespace cusort {
+
+  /**
+   * @brief  Sort key value pairs distributed across multiple GPUs
+   *
+   *   This sort function takes arrays of keys and values distributed
```

Comment:
We could add more documentation, for instance :
- if this is expected to be called inside a parallel section or by a single thread
- are there any assumptions such as the Cuda device being is already set to a specific id ?
- does this work only if the gpus are fully connected (ie each device must enable cuda access to all peers)?

I would recommend addressing this in the description so that users can double check this before filling issues.

Reply:
I will add some additional documentation tomorrow - but to answer your questions:
1. This is expected to be called by a single thread.  It uses openmp to start threads on each GPU
1. No assumptions on the cuda device, the openmp parallel region sets the device
1. The assumption here is SNMG.  We do cudaMemcpy calls (device to device) to transfer data between GPUs.  The function assumes that the peer access has already been enabled.

Reply:
There is no requirements that the GPUs be fully connected for the algorithm to work. However, performance will probably go down if they are not fully connected.

Reply:
Ah yes I think mentioning that peer access must be enabled by the user before calling this function is very important.